aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cu

```diff
@@ -94,7 +94,7 @@ SparseTensor coalesce_sparse_cuda(const SparseTensor& self) {
     int64_t stride = at::prod_intlist(values.sizes().slice(1));
     dim3 grid(THCCeilDiv(newNnz, (int64_t) SZ), THCCeilDiv(stride, (int64_t) C10_WARP_SIZE*SZ));
     dim3 block(C10_WARP_SIZE, SZ);
-    AT_DISPATCH_ALL_TYPES_AND2(
+    AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2(
```

Comment:
please add tests!

Reply:
All test for coalesce and uncoalesce cases with CPU and CUDA are covered in these new classes. 
`TestSparseComplex, ``TestUncoalescedSparseComplex`, `TestCudaSparseComplex`, `TestCudaUncoalescedSparseComplex`, and `TestUncoalescedSparse`
https://github.com/pytorch/pytorch/blob/5550442f7392a11c8f555109f33e884f30353b9b/test/test_sparse.py#L3203